src/shared/state-selectors/history-selectors.js

```diff
@@ -0,0 +1,25 @@
+import { createSelectorCreator, defaultMemoize } from "reselect";
```

Comment:
out of curiosity, is there a reason you went with `reselect` and not the native [`areStatesEqual`](https://react-redux.js.org/api/connect#arestatesequal-next-object-prev-object-boolean) in redux? I can see the value in reselect more broadly, but we're only using it in one place in the entire app here, and `areStatesEqual` feels like it can do the same thing without the added dependency. I don't think Redux performance is _across the board_ a problem (and indeed this PR greatly improves eval performance!), and places like history (and chunks) are the two heaviest part of the stores.

Reply:
i looked into this, and not including reselect at this point is fine with me. nice tip :-)